ratelimits/names.go

```diff
@@ -199,7 +200,7 @@ func validateFQDNSet(id string) error {
 		return fmt.Errorf(
 			"invalid fqdnSet, %q must be formatted 'fqdnSet'", id)
 	}
-	return policy.WellFormedDomainNames(domains)
+	return policy.WellFormedIdentifiers(identifier.SliceNewDNS(domains))
```

Comment:
Hmm, this is an issue. Because validateFQDNSet is going to be given a comma-separated list of already-stringified identifiers. They won't necessarily all be DNS identifiers, so calling SliceNewDNS isn't the right move. But they've also lost all identifying information, so reconstructing their actual identifier types is nontrivial (we don't want to be parsing strings to see whether they "look like" an IP vs a DNS name).

Honestly it's slightly odd that our rate limits code enforces just-as-strict rules on what we turn into redis keys as the RA enforces on what we include in certs. It should be _fine_ to construct a rate limit key even if one of the names in it is "invalid.*.example.org"? Worth talking through with @beautifulentropy

Reply:
Notes from our convo: Three options:
- Remove the rate limit package's dependency on WellFormedIdentifiers, because maybe it's not providing enough value to be worth keeping. This is clean, but hard to evaluate.
- Plumb identifiers through the ratelimits package, and change the fqdnset computation to include the identifier type. This would cause us to drop all of our current fqdnset rate limits when their keys change, but that's not the worst thing in the world.
- Same as above, but include a special case for "if all the idents are TypeDNS, use the old fqdnset computation method". This is currently my favorite option, as we can easily convert it to the option above by ripping out the conditional at a later date.